webpack/assets/javascripts/react_app/redux/API/APIRequest.js

```diff
@@ -1,24 +1,59 @@
 import { API } from './';
+import { actionTypeGenerator } from './APIActionTypeGenerator';
+import { STATUS } from '../../constants';
+import {
+  formatSuccess as defaultFormatSuccess,
+  formatFailure as defaultFormatFailure,
+  formatPending as defaultFormatPending,
+} from './APIHelpers';
 
-export const get = async (payload, url, store, actionTypes) => {
-  store.dispatch({
-    type: actionTypes.REQUEST,
+export const get = async (action, store) => {
+  const {
+    key,
+    url,
     payload,
+    actionTypes,
+    formatPending = defaultFormatPending,
+    formatSuccess = defaultFormatSuccess,
+    formatFailure = defaultFormatFailure,
+  } = action;
+  const { REQUEST, SUCCESS, FAILURE } = actionTypeGenerator(key, actionTypes);
+
+  store.dispatch({
+    key,
+    type: REQUEST,
+    payload: formatPending({
+      error: null,
```

Comment:
If error is `null` I don't see a reason to include the `error` key?

Reply:
same as the response here: https://github.com/theforeman/foreman/pull/7222#discussion_r358175598
plus it makes it easier to consume in the reducer:
```js
switch (type) {
    case REQUEST:
    case SUCCESS:
    case FAILURE:
      return state.merge({ [key]: { ...state[key], ...payload } });
    default:
      return state;
  }
```